### PR TITLE
Support pass prompt to CreateAlertTool

### PR DESCRIPTION
**Pull request by qianheng-aws**
### Description
This PR enhance the CreateAlertTool with 2 features:
- support pass prompt as parameter to override the default prompt.
- use CLAUDE  as its default model type if model_type in parameters is empty or unsupported.

### Related Issues
Resolves https://github.com/opensearch-project/skills/issues/447

### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/skills/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/agent/tools/CreateAlertTool.java [hunk 1]
```diff
@@ -62,29 +62,35 @@ public class CreateAlertTool implements Tool {
...
-                LoggerMessageFormat
-                    .format(
-                        null,
-                        "Failed to find the right prompt for modelType: {}, this tool supports prompts for these models: [{}]",
-                        modelType,
-                        String.join(",", promptDict.keySet())
-                    )
-            );
+        if (prompt.isEmpty()) {
+            if (!promptDict.containsKey(modelType)) {
```
Comment (yuye-aws) [inline src/main/java/org/opensearch/agent/tools/CreateAlertTool.java:79]: We can follow the behavior of PPL Tool. If modelType is an empty string `""`, then take its default value.

Comment (yuye-aws) [reply src/main/java/org/opensearch/agent/tools/CreateAlertTool.java:79]: FYI: https://github.com/opensearch-project/skills/blob/d0c42a8df6b43794f25f65e70f013695173582cd/src/main/java/org/opensearch/agent/tools/PPLTool.java#L136

**Review by yuye-aws — APPROVED** [review]
Approved. Can @zane-neo review?

File: src/main/java/org/opensearch/agent/tools/CreateAlertTool.java [hunk 2]
```diff
@@ -62,29 +63,53 @@ public class CreateAlertTool implements Tool {
...
+                return ModelType.CLAUDE;
+            }
+            try {
+                return ModelType.valueOf(value.toUpperCase(Locale.ROOT));
+            } catch (Exception e) {
+                log.error("Wrong Model type, should be CLAUDE or OPENAI");
+                return ModelType.CLAUDE;
+            }
+        }
+    }
```
Comment (yuye-aws) [inline src/main/java/org/opensearch/agent/tools/CreateAlertTool.java:91]: Wait a bit. The code is the same as the ModelType in PPL tool. Can you extract it outside the function for code reusage?

Comment (qianheng-aws) [reply src/main/java/org/opensearch/agent/tools/CreateAlertTool.java:91]: They are not exactly the same,  this tool doesn't support `FINETUNE` model. If we want to reuse this method, we'd better refactor it as well and I think it should also applies to other tools like `CreateAnomalyDetectorTool`.

So I don't think it's a blocker to this PR.

Comment (yuye-aws) [reply src/main/java/org/opensearch/agent/tools/CreateAlertTool.java:91]: I see. Then the code is good.

**Review by yuye-aws — APPROVED** [review]
This PR looks good to me. Pinging @xinyual to review again.

**Comment by yuye-aws** [comment]
https://github.com/opensearch-project/skills/pull/457 @qianheng-aws This PR gets merged. Please update accordingly.

**Comment by qianheng-aws** [comment]
> #457 @qianheng-aws This PR gets merged. Please update accordingly.

Should this PR be the same as that change? I think there is no difference functionally.

**Comment by yuye-aws** [comment]
> > #457 @qianheng-aws This PR gets merged. Please update accordingly.
> 
> Should this PR be the same as that change? I think there is no difference functionally. We already support empty model_type and return a default one.

Well. I mean you can make the following code snippet public. Then, both CreateAlertTool and CreateAnomalyDetectorTool can import it.

```
    public enum ModelType {
        CLAUDE,
        OPENAI;

        public static ModelType from(String value) {
            if (value.isEmpty()) {
                return ModelType.CLAUDE;
            }
            try {
                return ModelType.valueOf(value.toUpperCase(Locale.ROOT));
            } catch (Exception e) {
                log.error("Wrong Model type, should be CLAUDE or OPENAI");
                return ModelType.CLAUDE;
            }
        }
    }
```

**Comment by qianheng-aws** [comment]
> > > #457 @qianheng-aws This PR gets merged. Please update accordingly.
> > 
> > 
> > Should this PR be the same as that change? I think there is no difference functionally. We already support empty model_type and return a default one.
> 
> Well. I mean you can make the following code snippet public. Then, both CreateAlertTool and CreateAnomalyDetectorTool can import it.
> 
> ```
>     public enum ModelType {
>         CLAUDE,
>         OPENAI;
> 
>         public static ModelType from(String value) {
>             if (value.isEmpty()) {
>                 return ModelType.CLAUDE;
>             }
>             try {
>                 return ModelType.valueOf(value.toUpperCase(Locale.ROOT));
>             } catch (Exception e) {
>                 log.error("Wrong Model type, should be CLAUDE or OPENAI");
>                 return ModelType.CLAUDE;
>             }
>         }
>     }
> ```

I think that change should be made in the PR for refactoring code reuse. There is no help make it public for now.

**Comment by xinyual** [comment]
@qianheng-aws Seems CI/CD fails. Can you fix this?

**Comment by yuye-aws** [comment]
@qianheng-aws Spotless apply plz.

**Comment by qianheng-aws** [comment]
Only 1 flakey test failure, which is unrelated as well:

71 tests completed, 1 failed
Tests with failures:
 - org.opensearch.integTest.NeuralSparseSearchToolIT.testNeuralSparseSearchToolInFlowAgent_withNestedIndex

**Review by xinyual — APPROVED** [review]

**Review by yuye-aws — APPROVED** [review]

**Comment by yuye-aws** [comment]
@xinyual Plz attach backport-2.x label to this PR.

**Pull request merged**